Firestore/core/test/firebase/firestore/testutil/testutil.h

```diff
@@ -130,11 +134,13 @@ inline model::FieldValue::Map AddPairs(const model::FieldValue::Map& prior) {
  * @return The resulting map.
  */
 template <typename ValueType, typename... Args>
-model::FieldValue::Map AddPairs(const model::FieldValue::Map& prior,
-                                const std::string& key,
-                                const ValueType& value,
-                                Args... rest) {
-  return AddPairs(prior.insert(key, Value(value)), rest...);
+model::ObjectValue AddPairs(const model::ObjectValue& prior,
+                            const std::string& key,
+                            const ValueType& value,
+                            Args... rest) {
+  return AddPairs(
+      prior.Set(model::FieldPath::FromDotSeparatedString(key), Value(value)),
```

Comment:
Key isn't a dot-separated string here. It's the literal value of the key.

In general, the way these helpers are supposed to work is that they represent, as closely as possible, directly what's expected.

Secondarily this makes me uncomfortable to write tests for `FieldValue` using these primitives when they're implemented in terms of `FieldValue` itself. The tests are more rigorous if the system under test and the test helpers aren't built the same way.

Reply:
Oops; yeah the dot separator thing was because I was looking for the quickest way to turn a string into a fieldpath and forgot to revisit. In any case, it's no longer relevant.

Secondary concern addressed though. This now operates on FieldValue::Map (i.e. immutable::SortedMap) again.